_includes/code-carousel.html

```diff
@@ -0,0 +1,68 @@
+{% assign firstOption = true %}
+{% assign darkClassOpt = '' %}
+{% if page.darkmodeCode %}
+  {% assign darkClassOpt = 'dark' %}
+{% endif %}
+<div id="{{include.id}}" class="code-carousel">
+  {% assign totalSize = 0 %}
+  {% for example in include.collection %}
+  <input
+    type="radio"
+    name="{{include.id}}_tabs"
+    id="{{example.optionId}}_description"
+    data-target="{{example.optionId}}"
+    class="code-carousel_control"
+    hidden
+    aria-hidden="true"
+    {% if firstOption %}checked{% endif %}
+  >
+  {% assign firstOption = false %}
+  {% assign totalSize = totalSize | plus: 1 %}
+  {% endfor %}
+  <div class="content-area">
+    {% for example in include.collection %}
+    <section id="{{example.optionId}}_content">
+      <div class="code-element {{darkClassOpt}}">
+        <div class="bar-code"><span>{{example.codeTitle}}</span></div>
+        <div class="code-background {{darkClassOpt}} code-snippet-buttons">
+          <button class="code-snippet-button">
+            <a href="{{ example.scastieLink }}" target="_blank">
+              Run in playground <i class="fa-solid fa-play"></i>
+            </a>
+          </button>
+        </div>
+        <div class="code-snippet-area">
+          <div class="progress-label code-background {{darkClassOpt}}">
+            {{forloop.index}}<span class="progress-out-of"> /{{totalSize}}</span>
+          </div>
+          {% if forloop.first %}
+            {% assign previousIdx = include.collection.length | minus: 1 %}
+          {% else %}
+            {% assign previousIdx = forloop.index0 | minus: 1 %}
+          {% endif %}
+          {% if forloop.last %}
+            {% assign nextIdx = 0 %}
+          {% else %}
+            {% assign nextIdx = forloop.index %}
+          {% endif %}
+          <div class="arrow-switcher arrow-switcher_left {{darkClassOpt}}">
+            <label for="{{include.collection[previousIdx].optionId}}_description">
+              <i class="fa-solid fa-chevron-left"></i>
+            </label>
+          </div>
+          <div class="arrow-switcher arrow-switcher_right {{darkClassOpt}}">
+            <label for="{{include.collection[nextIdx].optionId}}_description">
+              <i class="fa-solid fa-chevron-right"></i>
+            </label>
+          </div>
+          <div class="code-background {{darkClassOpt}}">
+            <!-- the code snippet should be a raw markdown code block -->
+            {{example.content}}
+          </div>
+        </div>
+      </div>
+      <p>{{example.description}}</p>
```

Comment:
On my mobile, the description is half-hidden by the code:

![Screenshot_20230201-103842_Firefox](https://user-images.githubusercontent.com/332812/216007562-a17280eb-ccfb-45ec-88c5-ea7ad844c0f6.jpg)

(well, it looks like the code goes beyond its box)

Reply:
does it stay like this? I noticed that text can takes a while to scale and then shrink

Reply:
Does it eventually shrink after just waiting a few seconds? In my case, it didn’t seem to change, but I can try again…

Reply:
could you try?

Reply:
I can reproduce the overflow by setting the font size to large so ill try to make everything else scale appropriately

Reply:
I fixed it with https://github.com/scala/scala-lang/pull/1462/commits/d17c14d01582d7afd41386d54cfbcbe59c64ff84, just now have to make sure the code snippets are all the same line count

Reply:
The problem has gone with your latest changes.